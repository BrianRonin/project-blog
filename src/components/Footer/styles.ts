import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    background-color: ${theme.colors.white};
    a {
      color: ${theme.colors.darkText};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
