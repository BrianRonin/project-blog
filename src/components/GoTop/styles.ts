import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
