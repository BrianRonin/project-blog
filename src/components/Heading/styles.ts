import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
