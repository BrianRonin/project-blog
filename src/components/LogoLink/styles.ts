import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
