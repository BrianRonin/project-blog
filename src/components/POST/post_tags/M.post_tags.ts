import { postTagsProps } from './post_tags'

export const mock_post_tags = {
  tags: [
    {
      name: 'React',
      slug: 'react',
    },
    {
      name: 'Typescript',
      slug: 'typescript',
    },
    {
      name: 'Docker',
      slug: 'docker',
    },
    {
      name: 'Phyton',
      slug: 'phyton',
    },
    {
      name: 'JavaScript',
      slug: 'java-script',
    },
  ],
} as postTagsProps
