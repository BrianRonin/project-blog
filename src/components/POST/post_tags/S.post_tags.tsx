import styled, {
  css,
  DefaultTheme,
} from 'styled-components'
import { postTagsProps } from './post_tags'
type S_postTagsProps = postTagsProps

export const Main = styled.p<S_postTagsProps>`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium};

    span {
      margin: 0 0 0 0.5rem;
    }

    span::after {
      content: ', ';
    }

    span::before {
      content: ' ';
    }

    span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`
