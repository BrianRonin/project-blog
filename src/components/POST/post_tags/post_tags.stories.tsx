import {
  Meta,
  Story,
} from '@storybook/react/types-6-0'
import {
  PostTags,
  postTagsProps,
} from './post_tags'
import { mock_post_tags } from './M.post_tags'

export default {
  title: 'components/post/post tags',
  component: PostTags,
  args: mock_post_tags,
} as Meta

export const Template: Story<postTagsProps> = (
  args,
) => (
  <div style={{ padding: '2.4rem' }}>
    <p>
      Lorem ipsum dolor sit amet consectetur
      adipisicing elit. Debitis commodi
      voluptatibus culpa, placeat consectetur
      rerum hic id laborum error necessitatibus
      reprehenderit adipisci ipsum modi cumque
      odit iusto magnam non minima!
    </p>
    <PostTags {...args} />
    <p>
      Lorem ipsum dolor sit amet, consectetur
      adipisicing elit. Veritatis, sunt in? Iure
      laboriosam consectetur commodi! Pariatur
      exercitationem deserunt distinctio iste
      adipisci a in saepe quia consequatur dolor?
      Facere, nobis ex.
    </p>
  </div>
)
Template.parameters = {
  layout: 'fullscren',
  backgrounds: {
    default: 'light',
  },
}
