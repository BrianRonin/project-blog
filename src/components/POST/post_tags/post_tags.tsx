import Link from 'next/link'
import { type_strapi_tags } from '~/types/strapi/tags'
import * as S from './S.post_tags'

export type postTagsProps = {
  tags?: type_strapi_tags[]
}

export const PostTags = ({
  tags,
}: postTagsProps) => {
  if (tags?.length === 0) {
    return null
  }
  return (
    <S.Main>
      tags:
      {tags?.map((tag, index) => (
        <span key={`tag_reac_id_${index}`}>
          <Link href={`/tag/${tag.slug}`}>
            <a href=''>{tag.name}</a>
          </Link>
        </span>
      ))}
    </S.Main>
  )
}
