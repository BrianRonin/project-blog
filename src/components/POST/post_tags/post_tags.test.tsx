import { renderTheme } from '../../../styles/render-theme'
import { PostTags } from './post_tags'
import { mock_post_tags } from './M.post_tags'
import { screen } from '@testing-library/react'

describe('<PostTags />', () => {
  it('should render and snapshot', () => {
    const { container } = renderTheme(
      <PostTags {...mock_post_tags} />,
    )
    expect(
      screen.getByText(/Tags:/i),
    ).toBeInTheDocument()
    expect(
      screen.getAllByRole('link'),
    ).toHaveLength(5)
    expect(container).toMatchSnapshot()
  })
  it('should snapshot without tags', () => {
    const { container } = renderTheme(
      <PostTags
        {...mock_post_tags}
        tags={undefined}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
