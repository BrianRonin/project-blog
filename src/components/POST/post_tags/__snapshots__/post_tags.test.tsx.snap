// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should render and snapshot 1`] = `
.c0 {
  margin: 2.4rem;
}

.c0 span {
  margin: 0 0 0 0.5rem;
}

.c0 span::after {
  content: ', ';
}

.c0 span::before {
  content: ' ';
}

.c0 span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <p
    class="c0"
  >
    tags:
    <span>
      <a
        href=""
      >
        React
      </a>
    </span>
    <span>
      <a
        href=""
      >
        Typescript
      </a>
    </span>
    <span>
      <a
        href=""
      >
        Docker
      </a>
    </span>
    <span>
      <a
        href=""
      >
        Phyton
      </a>
    </span>
    <span>
      <a
        href=""
      >
        JavaScript
      </a>
    </span>
  </p>
</div>
`;

exports[`<PostTags /> should snapshot without tags 1`] = `
.c0 {
  margin: 2.4rem;
}

.c0 span {
  margin: 0 0 0 0.5rem;
}

.c0 span::after {
  content: ', ';
}

.c0 span::before {
  content: ' ';
}

.c0 span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <p
    class="c0"
  >
    tags:
  </p>
</div>
`;
